examples/async/native-async-mock/__tests__/native-async-mock-test.js

```diff
@@ -0,0 +1,11 @@
+// Copyright 2004-present Facebook. All Rights Reserved.
+
+'use strict';
+
+jest.mock('../native');
+const native = require('../native');
+
+test('mock works with native async', () => {
+  expect(native.asyncMethod).toBeDefined();
```

Comment:
So actually, the problem here is not with defining implementation of mock (which works well in both scenarios) but rather with method availability. When a method is an AsyncFunction type, its name isn't copied to the mock, but it does for sync methods, so I figured out we should follow this pattern. Does it make sense?